cmd/cspc-operator/app/operations.go

```diff
@@ -249,6 +250,10 @@ func (pc *PoolConfig) ClaimBD(bdName string) error {
 	if err != nil {
 		return errors.Wrapf(err, "could not get bd object %s", bdName)
 	}
+	// If blockdevice is already claimed no need of creating claim
+	if bdObj.Status.ClaimState == ndmapis.BlockDeviceClaimed {
+		return nil
+	}
```

Comment:
Commenting just to keep a track -- Either this new code or the piece of code in `claimBD` function needs to be removed.
Can be done as part of refactoring.